src/database/connection.ts

```diff
@@ -3,46 +3,93 @@ import waitForMySql from 'wait-for-mysql';
 import bluebird from 'bluebird';
 import { createConnection } from 'typeorm';
 import { isProd } from '@hollowverse/utils/helpers/env';
+import { readAwsSecretStringForStage } from '@hollowverse/utils/helpers/readAwsSecretStringForStage';
 import { entities } from './entities';
 import { stripIndent } from 'common-tags';
 import loggy from 'loggy';
 
-const {
-  TYPEORM_DATABASE,
-  TYPEORM_HOST,
-  TYPEORM_PASSWORD,
-  TYPEORM_USERNAME,
-  TYPEORM_PORT,
-} = process.env;
+type AwsSecretsManagerDbConfig = {
+  username: string;
+  password: string;
+  engine: string;
+  host: string;
+  port: number;
+  dbname: string;
+  dbInstanceIdentifier: string;
+};
 
-const waitAndPrintInstructions = async () => {
-  let hasConnected = false;
+const getDatabaseConfig = async () => {
+  if (process.env.STAGE === undefined || process.env.STAGE === 'local') {
+    const {
+      TYPEORM_DATABASE,
+      TYPEORM_HOST,
+      TYPEORM_PASSWORD,
```

Comment:
Potentially stupid question: wasn't the idea of using AWS Secrets Manager is to not have to pass secrets as env variables?

Reply:
The goal was to not have the secrets stored in the build artifacts. For local development, we do not expect the user to have the secret. The helper function `readSecretStringForStage` detects the environment and tries to use a fallback value if an env variable is used. This is an exception however, because the secret is stored as JSON and it would be inconvenient to pass a single env variable whose value is a serialized JSON string. So we handle the fallback value with multiple env variables.